include/eve/arch/cpu/wide.hpp

```diff
@@ -869,45 +885,73 @@ inline namespace EVE_ABI_NAMESPACE
     // Logical operations
     //==============================================================================================
     //! @brief Element-wise equality comparison of two eve::wide
-    friend EVE_FORCEINLINE logical<wide> operator==(wide v, wide w) noexcept
+    friend  EVE_FORCEINLINE auto operator==(wide const& v, wide const& w) noexcept
     {
       return detail::self_eq(v,w);
     }
 
     //! @brief Element-wise equality comparison of a eve::wide and a scalar value
     template<scalar_value S>
-    friend EVE_FORCEINLINE logical<wide> operator==(wide v, S w) noexcept
+    friend EVE_FORCEINLINE auto operator==(wide const& v, S w) noexcept
     {
       return v == wide{w};
     }
 
+    template<scalar_value S>
+    friend EVE_FORCEINLINE auto operator==(wide const& v, S w) noexcept
+            requires(detail::tag_dispatchable<tag::is_equal_,wide,S>)
+    {
+      return tagged_dispatch(tag::is_equal_{}, v, w);
+    }
+
     //! @brief Element-wise equality comparison of a scalar value and a eve::wide
     template<scalar_value S>
-    friend EVE_FORCEINLINE logical<wide> operator==(S v, wide w) noexcept
+    friend EVE_FORCEINLINE auto operator==(S v, wide const& w) noexcept
     {
       return w == v;
     }
 
+    template<scalar_value S>
+    friend EVE_FORCEINLINE auto operator==(S v, wide const& w) noexcept
```

Comment:
Do you want equal/not_equal to be enabled through tagged_dispatch?

I thought we'd go with:

```
struct point : kumi::tuple<int, int, int> {
   friend decltype(auto) x(eve::compatible<point> auto&& self) ...
   friend decltype(auto) y(eve::compatible<point> auto&& self) ...
   friend decltype(auto) z(eve::compatible<point> auto&& self) ...
   
   auto operator==(eve::compatible<point> auto const& a, eve::compatible<point> auto const& b) {
      return x(a) == x(b) && y(a) == y(b) && z(a) == z(b);
   } 
};
```

Reply:
(whatever you want to call the concept)

Reply:
As per the slack discussion, we want to limit the amount of APi for specialisation. I decided `tagged_dispatch` was a good vessel for that.